### PR TITLE
Regression Test repo pull request

### PR DESCRIPTION
**Pull request by asyatrhl**
Regression test codes are created for develop branch of MaximIntegratedAI/ai8x-training repository.

### CODE REVIEW

**Review by rotx-maxim — CHANGES_REQUESTED** [review]
See email for details

File: regression/test_config.yaml [hunk 1]
```diff
@@ -88,7 +88,7 @@ PascalVOC_2007_2012_256_320_augmented:
     threshold: -500
     epoch: 15
 Kinetics400:
-  datapath: "/data_ssd"
+  datapath: "/data/alican_data/"
```
Comment (rotx-maxim) [inline regression/test_config.yaml:91]: Use a generic destination and update documentation

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      - name: Activate Venv
+        run: source venv/bin/activate
+      - name: Install Dependencies
+        run: |
+          pip3 install -U pip wheel setuptools
+          pip3 install -r requirements-cu11.txt
+      - name: Create Test Script
+        run: python ./regression/create_test_script.py --testconf ./regression/test_config.yaml --testpaths ./regression/paths.yaml
+      - name: Run Training Scripts
+        run: bash /home/test/actions-runner/_work/ai8x-training/ai8x-training/scripts/output_file.sh
```
Comment (rotx-maxim) [inline .github/workflows/test.yml:62]: where does _work originate?

File: regression/create_onnx_script.py [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+    pathconfig = yaml.safe_load(file2)
+
+if not config["Onnx_Status"]:
+    sys.exit(1)
+
+folder_path = pathconfig["folder_path"]
+output_file_path = pathconfig["output_file_path_onnx"]
+train_path = pathconfig["train_path"]
+
+logs_list = folder_path + '/' + sorted(os.listdir(folder_path))[-1]
```
Comment (rotx-maxim) [inline regression/create_onnx_script.py:62]: should this be os.path.join()?

File: regression/create_onnx_script.py [hunk 4]
```diff
@@ -0,0 +1,130 @@
...
+        if "--use-bias" in line:
+            bias.append("--use-bias")
+        else:
+            bias.append("")
+
+    for file_p in sorted(os.listdir(logs_list)):
+        temp_path = logs_list + "/" + file_p
+        for temp_file in sorted(os.listdir(temp_path)):
+            if temp_file.endswith("_checkpoint.pth.tar"):
+                temp = f"{temp_path}/{temp_file}"
```
Comment (rotx-maxim) [inline regression/create_onnx_script.py:101]: os.path.join()?

File: regression/last_dev.py [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+            repo.remotes.origin.pull("develop")
+
+            dev_scripts(script_path, output_file_path)
+            cmd_command = "bash " + output_file_path
+            subprocess.run(cmd_command, shell=True, check=True)
+
+            path_command = "cd " + local_path
+            subprocess.run(path_command, shell=True, check=True)
+
+            source_path = "/home/asyaturhal/actions-runner/_work/ai8x-training/ai8x-training/logs/"
```
Comment (rotx-maxim) [inline regression/last_dev.py:146]: Should not use a hard coded personal home here

File: regression/log_comparison.py [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+
+for files_new in sorted(os.listdir(new_logs_path)):
+    files_new_temp = files_new.split("___")[0]
+    if files_new_temp not in old_log_list:
+        not_found_model.append(files_new_temp + " not found in last developed log files.")
+    for files_old in sorted(os.listdir(old_logs_path)):
+        files_old_temp = files_old.split("___")[0]
+        if files_old_temp == files_new_temp:
+
+            old_path = old_logs_path + '/' + files_old
```
Comment (rotx-maxim) [inline regression/log_comparison.py:202]: os.path.join() ?

File: regression/pass_fail.py [hunk 7]
```diff
@@ -0,0 +1,109 @@
...
+with open(yaml_path, 'r', encoding='utf-8') as file:
+    # Load the YAML content into a Python dictionary
+    config = yaml.safe_load(file)
+
+with open(test_path, 'r', encoding='utf-8') as file2:
+    # Load the YAML content into a Python dictionary
+    pathconfig = yaml.safe_load(file2)
+
+log_path = pathconfig["log_path"]
+log_path = log_path + '/' + sorted(os.listdir(log_path))[-1]
```
Comment (rotx-maxim) [inline regression/pass_fail.py:36]: os.path.join()

File: regression/pass_fail.py [hunk 8]
```diff
@@ -0,0 +1,109 @@
...
+for logs in sorted(os.listdir(log_path)):
+    log_name = (logs.split("___"))[0]
+    log_model = log_name.split("-")[0]
+    log_data = log_name.split("-")[1]
+
+    if log_data in config and log_model in config[log_data]:
+        threshold_temp = float(config[f'{log_data}'][f'{log_model}']['threshold'])
+    else:
+        threshold_temp = 0
+    logs = log_path + '/' + str(logs)
```
Comment (rotx-maxim) [inline regression/pass_fail.py:103]: ...

File: regression/test_config.yaml [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+  ai85tinierssd:
+    threshold: -500
+    epoch: 15
+PascalVOC_2007_2012_256_320_augmented:
+  datapath: "/data_ssd"
+  ai87fpndetector:
+    threshold: -500
+    epoch: 15
+Kinetics400:
+  datapath: "/data/alican_data/"
```
Comment (rotx-maxim) [inline regression/test_config.yaml:91]: avoid personal directories

**Review by rotx-maxim — CHANGES_REQUESTED** [review]

File: .github/workflows/test.yml (hunk 2 above)
Comment (asyatrhl) [reply .github/workflows/test.yml:62]: It is default working directory of the github runner. Name of this directory can be changed while installing for the MaximIntegratedAI:develop repo.

File: regression/create_onnx_script.py (hunk 3 above)
Comment (asyatrhl) [reply regression/create_onnx_script.py]: It is fixed.

File: regression/last_dev.py (hunk 5 above)
Comment (asyatrhl) [reply regression/last_dev.py]: It is fixed.

File: regression/create_onnx_script.py (hunk 4 above)
Comment (asyatrhl) [reply regression/create_onnx_script.py:101]: It is fixed.

File: regression/log_comparison.py (hunk 6 above)
Comment (asyatrhl) [reply regression/log_comparison.py]: It is fixed.

File: regression/pass_fail.py (hunk 7 above)
Comment (asyatrhl) [reply regression/pass_fail.py]: It is fixed.

File: regression/test_config.yaml (hunk 9 above)
Comment (asyatrhl) [reply regression/test_config.yaml]: It is fixed.

File: regression/test_config.yaml (hunk 1 above)
Comment (asyatrhl) [reply regression/test_config.yaml]: It is fixed.

**Review by ermanok — CHANGES_REQUESTED** [review]
It seems it is OK after a few minor updates I mentioned.

File: regression/create_test_script.py [hunk 10]
```diff
@@ -0,0 +1,102 @@
...
+
+parser = argparse.ArgumentParser()
+parser.add_argument('--testconf', help='Enter the config file for the test', required=True)
+parser.add_argument('--testpaths', help='Enter the paths for the test', required=True)
+args = parser.parse_args()
+yaml_path = args.testconf
+test_path = args.testpaths
+
+# Open the YAML file
+with open(yaml_path, 'r', encoding='utf-8') as file:
```
[the remark this answers is not in the document]
Comment (asyatrhl) [reply regression/create_test_script.py]: 'file' and 'file2' updated as 'yaml_file' and 'path_file'.

**Review by ermanok — APPROVED** [review]
Looks good.

**Review by rotx-maxim — APPROVED** [review]

**Pull request merged**